### ISSUE TITLE
IDE0004 false positive when using GetType()

### ISSUE THREAD

**Issue by nalka0** [dotnet/roslyn#75424]
**Version Used**: 

Visual Studio Community 2022 version 17.11.4

**Steps to Reproduce**:

```cs
var x = ((byte)3).GetType();
```

In the above snippet `x` represents `typeof(byte)` with the cast and would represent `typeof(int)` without it

**Diagnostic Id**:

`"IDE0004: Cast is redundant"`

**Expected Behavior**:

No IDE0004 on the above snippet

**Actual Behavior**:

IDE0004 happens on the above snippet
